app/Controllers/Http/CartsController.ts

```diff
@@ -0,0 +1,52 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import { schema } from '@ioc:Adonis/Core/Validator'
+import Product from 'App/Models/Product'
+
+export default class CartsController {
+  public async index({ auth }: HttpContextContract) {
+    const user = await auth.authenticate()
+    const carts = await user.related('carts').query().preload('product')
+
+    const response = carts.map((element) => {
+      return { product: element.product, quantity: element.quantity }
+    })
+
+    return response
+  }
+
+  public async store({ request, response, auth }: HttpContextContract) {
+    const { data } = await request.validate({
+      schema: schema.create({
+        data: schema.object().anyMembers(),
+      }),
+    })
+
+    const isEntryIncorrect = Object.entries(data).find(([key, value]) => {
+      const isEntryIncorrect = Math.sign(+key) !== 1 && Math.sign(value) !== 1
+      return isEntryIncorrect
+    })
```

Comment:
can just use nan instead of math.sign..... it returns boolean.... this will be easier to read

Reply:
nan will bypass zero and negative values... here key is id and value is quantity and both can't be negative